astropy/utils/tests/test_shapes.py

```diff
@@ -27,3 +30,37 @@ def test_unbroadcast():
     y = np.broadcast_to(x, (5, 3, 5))
     z = unbroadcast(y)
     assert z.shape == (3, 5)
+
+
+TEST_SHAPE = (13, 16, 4, 90)
+
+
+class TestSimplifyBasicIndex:
+    # We use a class here so that we can allocate the data once and for all to
+    # speed up the testing.
+
+    @pytest.fixture(autouse=True)
+    def setup_class(self, tmp_path):
+        self.shape = TEST_SHAPE
+        self.data = np.random.random(TEST_SHAPE)
+
+    @given(basic_indices(TEST_SHAPE))
+    def test_indexing(self, index):
+        new_index = simplify_basic_index(index, shape=self.shape)
+        assert_equal(self.data[index], self.data[new_index])
+        assert isinstance(new_index, tuple)
+        assert len(new_index) == len(self.shape)
+        for idim, idx in enumerate(new_index):
+            if isinstance(idx, slice):
+                assert isinstance(idx.start, int)
+                assert idx.start >= 0
+                assert idx.start < TEST_SHAPE[idim]
+                if idx.step is None or idx.step > 0:
```

Comment:
I think it would make sense to have no `slice(None)` at all here, so that the only thing to check would be `if idx.stop is not None`

Reply:
Done